src/hotspot/share/opto/node.cpp

```diff
@@ -619,30 +619,11 @@ void Node::destruct(PhaseValues* phase) {
   // See if the input array was allocated just prior to the object
   int edge_size = _max*sizeof(void*);
   int out_edge_size = _outmax*sizeof(void*);
-  char *edge_end = ((char*)_in) + edge_size;
+  char *in_array = ((char*)_in);
+  char *edge_end = in_array + edge_size;
   char *out_array = (char*)(_out == NO_OUT_ARRAY? NULL: _out);
   int node_size = size_of();
```

Comment:
This code should be moved down too before lines which trash fields.

Reply:
Moved to just below the ASSERT block that trashes fields.